estate_account/models/estate_property.py

```diff
@@ -0,0 +1,128 @@
+        :rtype: dict
+        """
+        self.ensure_one()
+
+        res = {
+            'display_type': 'product',
+            'sequence': sequence,
+            'name': self.env['account.move.line']._get_journal_items_full_name(self.name, 'House Sale'),
+            'quantity': 1,
+            'price_unit': self.selling_price,
+            'property_ids': [fields.Command.link(self.id)],
+        }
+
+        return res
+
+    # trying to understand sales invoice creation line by line
+    def _create_invoices(self, grouped=False, date=None):
+        if not self.env['account.move'].has_access('create'):
+            try:
+                self.check_access('write')
+            except AccessError:
+                return self.env['account.move']
+
+        invoice_vals_list = []
+        for invoice_item_sequence, record in enumerate(self):
+            if record.state != 'sold':
+                raise UserError('You can only invoice sold properties.')
+
+            if record.buyer_id.lang:
+                record = record.with_context(lang=record.buyer_id.lang)
+
+            invoice_vals = record._prepare_invoice()
+            invoice_vals['invoice_line_ids'].append(
+                fields.Command.create(record._prepare_invoice_line(sequence=invoice_item_sequence))
+            )
+            invoice_vals_list.append(invoice_vals)
+
+        if not grouped:
+            new_invoice_vals_list = []
+            invoice_grouping_keys = self._get_invoice_grouping_keys()
+            invoice_vals_list = sorted(
+                invoice_vals_list,
+                key=lambda x: [
+                    x.get(grouping_key) for grouping_key in invoice_grouping_keys
+                ]
+            )
+
+            for _grouping_keys, invoices in groupby(invoice_vals_list,
+                                                    key=lambda x: (x.get(grouping_key) for grouping_key in
+                                                                   invoice_grouping_keys)):
+                origins = set()
+                refs = set()
+                ref_invoice_vals = None
+                for invoice_vals in invoices:
+                    if not ref_invoice_vals:
+                        ref_invoice_vals = invoice_vals
+                    else:
+                        ref_invoice_vals['invoice_line_ids'] += invoice_vals['invoice_line_ids']
+                    origins.add(invoice_vals['invoice_origin'])
+                    refs.add(invoice_vals['ref'])
+
+                ref_invoice_vals.update({
+                    'ref': ', '.join(refs)[:2000],
+                    'invoice_origin': ', '.join(origins)
+                })
+                new_invoice_vals_list.append(ref_invoice_vals)
+            invoice_vals_list = new_invoice_vals_list
+
+            return self.env['account.move'].sudo().with_context(default_move_type='out_invoice').create(
+                invoice_vals_list)
```

Comment:
This whole method is way too complicated for our case. You can practically use the same example as shown in the tutorial to create a move with the correct invoice_line_ids ;)

Reply:
That would be nice indeed. You can then use a small method that can add a invoice line to an existing move, or put that simple code directly into the method where you create the move (assuming you change a bit the flow to allow creating a move after selecting multiple properties for example), but this code is still doing way too complicated (and useless in our case) things :P